src/io.d

```diff
@@ -133,14 +138,47 @@ struct Position
         writer.put(text.take(col - 1).map!(c => (c == '\t') ? c : ' '));
         writer.put('^');
     }
+
+    /*
+     * Reports a position with an offset in the form <file-name>@<offset> instead of line and column.
+     * This is a machine interface for allowing the Epslion language server to add markers via the offset. 
+     */
+    string toStringWithOffset() const @safe
+    {
+        import std.array : appender;
+
+        auto writer = appender!string;
+
+        toStringWithOffset(writer);
+        return writer[];
+    }
+
+    void toStringWithOffset(W)(ref W writer) const
+    if (isOutputRange!(W, char))
+    {
+        import std.algorithm : map;
+        import std.format : format;
+
+        if (this == UndefPos)
+        {
+            const link = format!"%s@0:1:1"(name);
+            writer.put(link);
+        }
+        else {
+            const link = format!"%s@%s:%s:%s"(name, offset, line, col);
```

Comment:
Do you still need `line` and `col` with the offset format?
I hoped that `abc.eag@42` would be enough.

Reply:
Originally, I thought, I would need them just to have redundant information (thinking, redundant information is never bad). But in fact, it is not really needed and can be omitted.
I will update the PR for that.